client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/OutboundLineEdit.tsx

```diff
@@ -1,197 +1,130 @@
-    useState(false);
-  useFocusNumberOfPacksInput(draft);
+  const { next, disabled: nextDisabled } = useNextItem(itemId);
 
-  const placeholder = draftStockOutLines?.find(
-    ({ type, numberOfPacks }) =>
-      type === InvoiceLineNodeType.UnallocatedStock && numberOfPacks !== 0
+  const { data: itemData, isFetching } = useOutboundLineEditData(
+    invoiceId,
+    itemId
   );
+  const { mutateAsync } = useSaveOutboundLines(invoiceId);
+  const { saveBarcode } = useOpenedWithBarcode(asBarcodeOrNull(openedWith));
 
-  const onUpdateQuantity = (batchId: string, quantity: number) => {
-    updateQuantity(batchId, quantity);
-    setIsAutoAllocated(false);
-    setIsDirty(true);
-  };
+  const {
+    draftLines,
+    allocatedUnits,
+    placeholderQuantity,
+    alerts,
+    isDirty,
+    setAlerts,
+    clear,
+  } = useAllocationContext(state => ({
+    draftLines: state.draftLines,
+    placeholderQuantity: state.placeholderQuantity,
+    allocatedUnits: getAllocatedUnits(state),
+    alerts: state.alerts,
+    isDirty: state.isDirty,
+    setAlerts: state.setAlerts,
+    clear: state.clear,
+  }));
 
   const onSave = async () => {
     if (!isDirty) return;
+    if (!itemId) return;
 
-    await mutateAsync(draftStockOutLines);
-    if (!draft) return;
-
-    const { barcode } = draft;
-    const barcodeExists = !!barcode?.id;
-
-    if (!barcode || !currentItem || barcodeExists) return;
-
-    // it is possible for the user to select multiple batch lines
-    // if the scanned barcode does not contain a batch number
-    // however the scanned barcode can only relate to a specific pack size and therefore batch
-    const packSize = draftStockOutLines.find(
-      line => line.numberOfPacks > 0
-    )?.packSize;
-
-    const input = {
-      input: {
-        gtin: barcode.gtin,
-        itemId: currentItem?.id,
-        packSize,
-      },
-    };
+    await mutateAsync({
+      lines: draftLines,
+      itemId,
+      placeholderQuantity,
+    });
 
     try {
-      await insertBarcode(input);
+      await saveBarcode(itemId);
     } catch (error) {
       warning(t('error.unable-to-save-barcode', { error }))();
     }
   };
 
-  const onAllocate = (
-    newVal: number,
-    packSize: number | null,
-    autoAllocated = false
-  ) => {
-    const newAllocateQuantities = allocateQuantities(
-      status,
-      draftStockOutLines
-    )(newVal, packSize, undefined);
-    setIsDirty(true);
-    setDraftStockOutLines(newAllocateQuantities ?? draftStockOutLines);
-    setIsAutoAllocated(autoAllocated);
-    if (showZeroQuantityConfirmation && newVal !== 0)
-      setShowZeroQuantityConfirmation(false);
-
-    return newAllocateQuantities;
-  };
-
-  const canAutoAllocate = !!(currentItem && draftStockOutLines.length);
-  const okNextDisabled =
-    (mode === ModalMode.Update && nextDisabled) || !currentItem;
+  const okNextDisabled = (mode === ModalMode.Update && nextDisabled) || !itemId;
```

Comment:
I was confused about this (although I think it's product decision, this behaviour will need to be listed to confirm that's what we want), in add new can ok next to new item but in update mode cant

Reply:
Added to carry over, behaviour to list and confirm